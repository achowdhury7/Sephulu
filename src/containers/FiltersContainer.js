import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { PanelGroup, Panel } from 'react-bootstrap'
import Collapsible from 'react-collapsible'

import { filterActions, productActions } from '../actions'
import { Header, PageContent, FilterCheckbox } from '../components'


class FiltersContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      categories: [],
      priceRanges: []
    }
    this.handleCheckboxChange = this.handleCheckboxChange.bind(this)
  }

  handleCheckboxChange(event) {
    const target = event.target
    if (target.checked) {
      this.props.actions.selectCategoryFilter(target.name)
    } else {
      this.props.actions.deselectCategoryFilter(target.name)
    }
  }

  render() {
    const categoryCheckBoxes = this.props.categories.map(category => (
      <FilterCheckbox 
        key={category} 
        label={category} 
        labelClass="filter-checkbox" 
        name={category} 
        onChange={this.handleCheckboxChange} />
    ))
    const priceRangeCheckboxes = this.props.priceRanges.map(priceRange => (
      <FilterCheckbox 
        labelClass="filter-checkbox" 
        key={priceRange.max + priceRange.min}
        label={`$${(priceRange.min/100).toFixed(2)}-$${(priceRange.max/100).toFixed(2)}`}
        name={`range${(priceRange.min/100).toFixed(2)}-${(priceRange.max/100).toFixed(2)}`} />
    ))
    return (
      <div>
        <Collapsible className="filter-heading" trigger="Categories">
          {categoryCheckBoxes}
        </Collapsible>
        <Collapsible className="filter-heading" trigger="Price-Range">
          {priceRangeCheckboxes}
        </Collapsible>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => 
  ({
    actions: bindActionCreators(Object.assign(
      {},
      filterActions,
      productActions),
      dispatch
    )
  })

const mapStateToProps = state => ({
  categories: state.app.categories,
  priceRanges: state.app.priceRanges
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FiltersContainer)

