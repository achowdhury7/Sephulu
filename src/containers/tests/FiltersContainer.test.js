import React from 'react'
import { shallow, mount, render } from 'enzyme'

import { FiltersContainer } from '../FiltersContainer'

let wrapper

beforeEach(() => {
  const mockCategories = ['a', 'b', 'c']
  const mockPriceRanges = [
    { min: 0, max: 10 },
    { min: 10, max: 20 },
    { min: 20, max: 30 },
  ]
  wrapper = shallow(
    <FiltersContainer categories={mockCategories} priceRanges={mockPriceRanges} />
  )
})

test('render filters', () => {  
  expect(wrapper).toMatchSnapshot()
})
