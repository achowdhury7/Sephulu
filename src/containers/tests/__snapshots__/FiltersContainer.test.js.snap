// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render filters 1`] = `
<div>
  <div
    className="filter-container"
  >
    <div
      className="filter-header"
    >
      Categories
    </div>
    <FilterCheckbox
      key="a"
      label="a"
      labelClass="filter-checkbox"
      name="a"
      onChange={[Function]}
    />
    <FilterCheckbox
      key="b"
      label="b"
      labelClass="filter-checkbox"
      name="b"
      onChange={[Function]}
    />
    <FilterCheckbox
      key="c"
      label="c"
      labelClass="filter-checkbox"
      name="c"
      onChange={[Function]}
    />
  </div>
  <div
    className="filter-container"
    trigger="Price-Range"
  >
    <div
      className="filter-header"
    >
      Price
    </div>
    <FilterCheckbox
      key="10"
      label="$0-$0"
      labelClass="filter-checkbox"
      name={0}
      onChange={[Function]}
    />
    <FilterCheckbox
      key="30"
      label="$1-$0"
      labelClass="filter-checkbox"
      name={10}
      onChange={[Function]}
    />
    <FilterCheckbox
      key="50"
      label="$1-$0"
      labelClass="filter-checkbox"
      name={20}
      onChange={[Function]}
    />
  </div>
</div>
`;
